internal/k8s/reconciler/route.go

```diff
@@ -371,8 +373,10 @@ func (r *K8sRoute) Validate(ctx context.Context) error {
 				if err != nil {
 					return err
 				} else if !allowed {
-					// FUTURE Set appropriate status and early out
-					r.logger.Warn("Cross-namespace routing not allowed, will enforce in future", "refName", ref.Name, "refNamespace", ref.Namespace)
+					msg := fmt.Sprintf("Cross-namespace routing not allowed without matching referencepolicy refName: %v refNamespace: %v", ref.Name, *ref.Namespace)
+					r.logger.Warn(msg)
```

Comment:
I'm used to logs getting piped into systems like Splunk where it's best to put the unique pieces of information in the metadata like this:
```go
r.logger.Warn("Cross-namespace routing not allowed without matching ReferencePolicy", "refName", ref.Name, "refNamespace", ref.Namespace)
```

I'm not sure if this is a pattern that HashiCorp uses but am curious if someone knows

Reply:
Yeah, so the `go-hclog` project we use everywhere (including for our loggers in this project) is a more or less lightweight structured logger, so we should definitely leverage that rather than constructing plain ole strings. Also, if someone turns on [`logJSON`](https://www.consul.io/docs/k8s/helm#v-global-logjson) in `helm` -- using the structured logs will move these into their own fields that can be easily filtered in a centralized log store as you mention.